code/modules/plumbing/plumbers/autohydro.dm

```diff
@@ -42,7 +41,7 @@
 
 /obj/machinery/hydroponics/constructable/automagic/process()
 	if(reagents)
-		applyChemicals(reagents)
+		reagents.add_reagent(reagents)
```

Comment:
does this do anything, even

Reply:
actually if you're going to add the plumbing functionality to base trays you may just want to remove most/all of the code here, put it in the basic hydrotray, and have a var or something control autoharvest (because that's the only thing these trays will have different, apart from being unable to add reagents manually)

Reply:
I think for now, I'll definitely remove the autohydro, and then look at how TG's got plumbing set up so I can tweak the tray. If I recall, I *think* TG has the hydrotrays now built to be automatically plumbing compatible as of recently, but I'm also hesitant to make a move on the whole plumbing system because on TG's side it's got a lot of kinks that need to be worked out.

Reply:
Maybe I can add an autoharvest function to the base tray too, though that will be dangerous for plants at high instability, because it's very well possible to have a plant roll two chems that react and then explode.